optaweb-employee-rostering-frontend/src/ui/pages/employee/EmployeesPage.tsx

```diff
@@ -66,9 +66,9 @@ const mapStateToProps = (state: AppState, ownProps: Props): StateProps => ({
     ownProps.t('shortId'),
     ownProps.t('color'),
   ],
-  tableData: employeeSelectors.getEmployeeList(state),
-  skillList: skillSelectors.getSkillList(state),
-  contractList: contractSelectors.getContractList(state),
+  tableData: employeeSelectors.getEmployeeList(state).toArray(),
+  skillList: skillSelectors.getSkillList(state).toArray(),
+  contractList: contractSelectors.getContractList(state).toArray(),
```

Comment:
@Christopher-Chianelli I think one of the questions from the other thread is about this. I would say let's not do this. The selector function should return exactly the data that the UI component needs (so the `toArray()` transformation should happen inside the selector).

I'm not 100% sure but I think about components using plain arrays like this: yes, we give up immutability but we get a simpler code (`employees[0]` vs `employees.get(0) ?? error()`). Correct me if I'm wrong but IMO we don't need immutability in components because they do not modify (redux) state directly (only reading props, dispatching actions).

Reply:
Correct. That question had two parts: 1) Should the selector return List or array, and 2) If array, should the selector clone it (providing local immutable at a performance cost) or cache a single result array (mutable but faster)

Reply:
Return mutable array, I would say.